subprojects/common/report-viewer/src/main/kotlin/com/avito/report/TestArtifactsProviderFactory.kt

```diff
@@ -0,0 +1,46 @@
+package com.avito.report
+
+import com.avito.report.internal.DirectTestArtifactsProvider
+import com.avito.report.internal.UniqueDirTestArtifactsProvider
+import com.avito.report.model.TestStaticData
+import java.io.File
+
+object TestArtifactsProviderFactory {
+
+    fun create(
+        testReportRootDir: Lazy<File>,
+        testStaticData: TestStaticData
+    ): TestArtifactsProvider {
+        return UniqueDirTestArtifactsProvider(
+            rootDir = testReportRootDir,
+            testDirGenerator = TestDirGenerator.StaticData(testStaticData)
+        )
+    }
+
+    fun create(
+        testReportRootDir: Lazy<File>,
+        className: String,
+        methodName: String
+    ): TestArtifactsProvider {
+        return UniqueDirTestArtifactsProvider(
+            rootDir = testReportRootDir,
+            testDirGenerator = TestDirGenerator.Impl(className, methodName)
+        )
+    }
+
+    fun createForTempDir(tempDir: File): TestArtifactsProvider {
+        return DirectTestArtifactsProvider(lazy { tempDir })
+    }
+
+    @Suppress("SdCardPath") // android API's are unavailable here
+    fun createForAdbAccess(appUnderTestPackage: String, className: String, methodName: String): TestArtifactsProvider {
```

Comment:
actual fix: access from different directories:

1. `ExternalStorageTransport` writes to `getExternalFilesDirs(targetContenxt)`
2. `DeviceWorker` specifies path on device via  `createForAdbAccess()` (to be more precise: `/sdcard/Android/data/${appUnderTestPackage}files`)
3. `ArtifactsTestListener` creates tempDirectory and pulls its content using path provided by `DeviceWorker`
4. `TestArtifactsProcessor` reads from tempDirectory and uploads multiple files in parallel but in blocking manner
5. `ArtifactsTestListener` removes tempDirectory after that